packages/quantic/force-app/main/default/lwc/quanticInsightSummary/__tests__/quanticInsightSummary.test.js

```diff
@@ -0,0 +1,225 @@
+};
+
+function prepareHeadlessState() {
+  // @ts-ignore
+  mockHeadlessLoader.getHeadlessBundle = () => {
+    return {
+      buildSearchBox: functionsMocks.buildSearchBox,
+      buildQuerySummary: functionsMocks.buildQuerySummary,
+    };
+  };
+}
+
+function mockSuccessfulHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element, _, initialize) => {
+    if (element instanceof QuanticInsightSummary && !isInitialized) {
+      isInitialized = true;
+      initialize(exampleEngine);
+    }
+  };
+}
+
+function mockErroneousHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element) => {
+    if (element instanceof QuanticInsightSummary) {
+      element.setInitializationError();
+    }
+  };
+}
+
+const createTestComponent = buildCreateTestComponent(
+  QuanticInsightSummary,
+  'c-quantic-insight-summary',
+  {defaultOptions}
+);
+
+describe('c-quantic-insight-summary', () => {
+  beforeEach(() => {
+    mockSuccessfulHeadlessInitialization();
+    prepareHeadlessState();
+  });
+
+  afterEach(() => {
+    cleanup();
+    searchBoxState = initialSearchBoxState;
+    querySummaryState = initialQuerySummaryState;
+    isInitialized = false;
+  });
+
+  describe('component initialization', () => {
+    it('should initialize the search box and query summary headless controllers', async () => {
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionsMocks.buildSearchBox).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.buildSearchBox).toHaveBeenCalledWith(exampleEngine);
+      expect(functionsMocks.buildQuerySummary).toHaveBeenCalledTimes(1);
+      expect(functionsMocks.buildQuerySummary).toHaveBeenCalledWith(
+        exampleEngine
+      );
+    });
+
+    it('should subscribe the search box and query summary state changes', async () => {
+      createTestComponent();
+      await flushPromises();
+    });
+  });
+
+  describe('component disconnection', () => {
+    it('should unsubscribe from the search box and query summary state changes', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      document.body.removeChild(element);
+
+      expect(functionsMocks.searchBoxStateUnsubscriber).toHaveBeenCalledTimes(
+        1
+      );
+      expect(
+        functionsMocks.querySummaryStateUnsubscriber
+      ).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('when an initialization error occurs', () => {
+    beforeEach(() => {
+      mockErroneousHeadlessInitialization();
+    });
+
+    it('should display the initialization error component', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const initializationError = element.shadowRoot.querySelector(
+        selectors.initializationError
+      );
+
+      expect(initializationError).not.toBeNull();
+    });
+  });
+
+  describe('when no search query is executed', () => {
```

Comment:
```suggestion
  describe('when the searchbox state value is empty', () => {
```
It's more descriptive to what the insight summary really does :)
Because a search query did still happen, it's just that the `q` was empty essentially.